File: internal/infrastructure/config/uuid_generator.go
package config

import (
	"github.com/google/uuid"
)

// UuidGenerator implements the IdGenerator contract using Google's UUID library
type UuidGenerator struct{}

// NewUuidGenerator creates a new UuidGenerator instance
func NewUuidGenerator() *UuidGenerator {
	return &UuidGenerator{}
}

// Generate creates a new UUID string
func (u *UuidGenerator) Generate() string {
	return uuid.New().String()
}

File: api/src/main/java/br/unifor/costify/shared/infra/config/UuidGenerator.java
package br.unifor.costify.shared.infra.config;

import br.unifor.costify.shared.domain.contracts.IdGenerator;
import java.util.UUID;
import org.springframework.stereotype.Component;

@Component
public class UuidGenerator implements IdGenerator {
  @Override
  public String generate() {
    return UUID.randomUUID().toString();
  }
}
